src/org/jitsi/impl/neomedia/MediaStreamImpl.java

```diff
@@ -3751,12 +3752,47 @@ public int getSpatialID(byte[] buf, int off, int len)
         }
         else
         {
-            logger.warn("Method not implemented");
+            // XXX not implementing temporal layer detection should not break
+            // things.
             return -1;
         }
     }
 
     /**
+     * Returns a boolean that indicates whether or not our we're able to detect
+     * the frame boundaries for the codec of the packet that is specified as an
+     * argument.
+     *
+     * @param baf the {@link ByteArrayBuffer} that holds the RTP packet.
+     *
+     * @return true if we're able to detect the frame boundaries for the codec
+     * of the packet that is specified as an argument, false otherwise.
+     */
+    public boolean supportsFrameBoundaries(ByteArrayBuffer baf)
+    {
+        if (frameMarkingsExtensionId == -1)
+        {
+            REDBlock redBlock = getPayloadBlock(baf);
+            if (redBlock != null && redBlock.getLength() != 0)
+            {
+                final byte vp9PT = getDynamicRTPPayloadType(Constants.VP9),
+                    vp8PT = getDynamicRTPPayloadType(Constants.VP8),
+                    pt = redBlock.getPayloadType();
+
+                return vp9PT == pt || vp8PT == pt;
```

Comment:
also wondering if there's a different place where we could do the codec comparison that might be more generic (a utility function somewhere or something)?